Physical_Engine/Query/MaximumRadius.cs

```diff
@@ -0,0 +1,122 @@
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Gets the maximum radius based on the diameter of the reinforcement bar using the values given in BS 8666:2020 Table 8.")]
+        [Input("diameter", "The diameter of the reinforcement bar to determine the maximum bending radius.", typeof(Length))]
+        [Output("maximumRadius", "The maximum scheduling radius based on the diameter of the reinforcement bar", typeof(Length))]
+        public static double MaximumRadius(this Reinforcement reinforcement)
+        {
+            return reinforcement.IsNull() ? 0 : MaximumRadius(reinforcement.ShapeCode, reinforcement.Diameter);
+        }
+
+        /***************************************************/
+
+        [Description("Gets the maximum radius based on the diameter of the reinforcement bar using the values given in BS 8666:2020 Table 8.")]
+        [Input("diameter", "The diameter of the reinforcement bar to determine the maximum bending radius.", typeof(Length))]
+        [Output("maximumRadius", "The maximum scheduling radius based on the diameter of the reinforcement bar", typeof(Length))]
+        public static double MaximumRadius(this IShapeCode shapeCode, double diameter)
+        {
+            if (diameter <= 0)
+            {
+                Reflection.Compute.RecordError("The diameter must be greater than 0. The maximum radius cannot be calculated.");
+                return 0;
+            }
+
+            string fullNamespace = shapeCode.GetType().Namespace;
+            int position = shapeCode.GetType().Namespace.LastIndexOf(".") + 1;
+            string standard = fullNamespace.Substring(position, fullNamespace.Length - position);
+
+            switch (standard)
+            {
+                case "BS8666":
+                    List<double> diameters = new List<double>() { 6, 8, 10, 12, 16, 20, 25, 32, 40 }.Select(x => x / 1000).ToList();
+                    List<double> radii = new List<double>() { 2.5, 2.75, 3.5, 4.25, 7.5, 14, 30, 43, 58 };
+
+                    Dictionary<double, double> maximumBendingRadius = diameters.Zip(radii, (k, v) => new { k, v }).ToDictionary(x => x.k, x => x.v);
+
+                    return maximumBendingRadius.LinearInterpolate(diameter);
+                default:
+                    Reflection.Compute.RecordError("Standard not recognised or supported, the scheduling radius could not be calculated.");
+                    return 0;
+            }
+        }
+
+        /***************************************************/
+        /**** Private Methods                            ****/
+        /***************************************************/
+
+        private static double LinearInterpolate(this Dictionary<double, double> reference, double x )
```

Comment:
It tempts me to ask for making it public and putting in a more central engine - not sure where, however, and not sure if the UI would lift it. So OK with keeping it as is.

Reply:
I assume it could be used elsewhere, but we can always promote it up later.